test-case/check-ipc-flood.sh

```diff
@@ -33,12 +34,12 @@ lpc_loop_cnt=${OPT_VAL['c']}
 ipc_flood_dfs=${OPT_VAL['f']}
 loop_cnt=${OPT_VAL['l']}
 
-[[ ! "$(sof-kernel-dump.sh|grep 'sof-audio'|grep 'Firmware debug build')" ]] && dlogw "${BASH_SOURCE[0]} need debug version firmware" && exit 2
+! sof-kernel-dump.sh | grep sof-audio | grep -q "Firmware debug" && dlogw "${BASH_SOURCE[0]} need debug version firmware" && exit 2
```

Comment:
Please take this opportunity to remove the negation, split the too long line and remove the hardcoded exit code

```shell
sof-kernel-dump.sh | grep sof-audio | grep -q "Firmware debug" ||
     skip "need debug version firmware"
```

Reply:
I was intentionally trying to do the minimum to resolve the shellcheck issues.  It also wouldn't have occurred to me to use skip at all, what does it do? I don't see it in lib.sh (just skip_test) but I don't think that presents as an error.

Reply:
> I was intentionally trying to do the minimum to resolve the shellcheck issues.

That's fair but you did change the code, I mean you removed `[[ ... ]]` so you already crossed the "code change" line, it's not just adding quotes to please shellcheck's paranoia about special characters.

Note I'm not requesting any changes outside the lines that you are already changing and re-testing. That's why I'm requesting a full clean-up of these lines (and only of these lines): because you can leverage the context and knowledge and test environment that you already have about these lines.

> It also wouldn't have occurred to me to use skip at all, what does it do? I don't see it in lib.sh (just skip_test) but I don't think that presents as an error.

Sorry I meant `skip_test`. It should behave the same without hardcoding the magic value `2`, please test and make sure.